### PR TITLE
fix(wallet): Add tab query param to keep selected tab state

### PR DESCRIPTION
**Pull request by muliswilliam**
Resolves: https://github.com/brave/brave-browser/issues/33307

## Submitter Checklist:

- [x] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:

https://github.com/brave/brave-core/assets/48117473/4f6a65c7-6980-4de4-80c2-c08ab96c8e3d

### CODE REVIEW

**Review by Douglashdaniel — APPROVED** [review]
++

File: components/brave_wallet_ui/common/hooks/use-query.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
+// Copyright (c) 2023 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+import * as React from 'react'
+import { useLocation } from 'react-router-dom'
+
+// Parse query string
```
Comment (josheleonard) [inline components/brave_wallet_ui/common/hooks/use-query.ts:9]: Use doc comment format

File: components/brave_wallet_ui/components/desktop/views/nfts/components/nfts.tsx [hunk 2]
```diff
@@ -110,13 +111,14 @@ export const Nfts = (props: Props) => {
...
     localStorage.getItem(LOCAL_STORAGE_KEYS.IS_ENABLE_NFT_AUTO_DISCOVERY_MODAL_HIDDEN) === null
   )
-  const [selectedOptionId, setSelectedOptionId] = React.useState<NftDropdownOptionId>('collected')
   const [showSearchBar, setShowSearchBar] = React.useState<boolean>(false)
 
   // hooks
   const history = useHistory()
   const dispatch = useDispatch()
   const { nonFungibleTokens, isIpfsBannerVisible, onToggleShowIpfsBanner } = useNftPin()
+  const tab = useQuery().get('tab')
```
Comment (josheleonard) [inline components/brave_wallet_ui/components/desktop/views/nfts/components/nfts.tsx:120]: Maybe call .get on a sperate line?

File: components/brave_wallet_ui/common/hooks/use-query.ts (hunk 1 above)
Comment (muliswilliam) [reply components/brave_wallet_ui/common/hooks/use-query.ts]: Added doc comment: ba5d4488d86f3ed191f7add0e95d9b14809d1f23

File: components/brave_wallet_ui/components/desktop/views/nfts/components/nfts.tsx (hunk 2 above)
Comment (muliswilliam) [reply components/brave_wallet_ui/components/desktop/views/nfts/components/nfts.tsx]: Call .get on a separate line: 5fbe56989f3fd0d990c6898669a84a7d1a0ed448

**Pull request merged**